### PR TITLE
Fix/1202812787027835 improve chain response time

### PR DESCRIPTION
**Pull request by irediaes**
* improved waiting time when requesting service; multsig wallet funding is now done in parallel with fetch did document from chain
*  improved waiting time when approving transactions; only spent transaction is waited until finalized

### CODE REVIEW

File: rust/Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 .DEFAULT_GOAL := help
 PROJECTNAME=peaq_codec_api
-LLVM_PATH=/usr/local/homebrew/opt/llvm
+# LLVM_PATH=/usr/local/homebrew/opt/llvm
+LLVM_PATH=/opt/homebrew/opt/llvm
```
Comment (sfffaaa) [inline rust/Makefile]: Does that mean when we build this project, we have to install the LLVM into this path, `/opt/homebrew/opt/llvm`?

File: lib/screens/home.dart [hunk 2]
```diff
@@ -166,19 +167,23 @@ class _HomeScreenState extends State<HomeScreen>
...
                                           }
                                           _dumbChargeProvider!.qrController
                                               .pause();
                                           _chargeProvider.providerDid = data;
+                                          _chargeProvider
+                                              .generateAndFundMultisigWallet();
+
                                           await _chargeProvider
                                               .fetchProviderDidDocument(data);
+                                          if (!mounted) return;
```
Comment (sfffaaa) [inline lib/screens/home.dart:186]: I didn't see any other place using this parameter. Do I miss something?

Comment (irediaes) [reply lib/screens/home.dart:186]: It's a flutter magic :) 
`mounted` is actually a flutter global scope variable that's inferred.

The code is saying:
if the the screen is not yet mounted, return.

File: rust/Makefile (hunk 1 above)
Comment (irediaes) [reply rust/Makefile]: LLVM is needed to build the android binaries;  getting the path from Env var is a better approach. I will do that now.

Comment (irediaes) [reply rust/Makefile]: Changed

**Review by sfffaaa — APPROVED** [review]

**Pull request merged**